Library/Homebrew/cmd/upgrade.rb

```diff
@@ -119,15 +119,19 @@ def upgrade_formula(f)
       tab = Tab.for_keg(keg)
     end
 
+    new_build_options = BuildOptions.new(Options.create(ARGV.flags_only), f.options)
```

Comment:
Just go for `build_options` to be consistent with `reinstall` (or rename it there too if you prefer this name)